components/raftstore/src/coprocessor/mod.rs

```diff
@@ -145,8 +145,8 @@ pub trait AdminObserver: Coprocessor {
         &self,
         _ctx: &mut ObserverContext<'_>,
         _tr: &TransferLeaderRequest,
-    ) -> Result<()> {
-        Ok(())
+    ) -> Result<Option<ExtraMessage>> {
```

Comment:
Shall we put `PollCtx::trans` into `ObserverContext` instead of change this function definition?

Reply:
What's your thought? @overvenus

Reply:
Let's keep the current implementation. Moving `trans` into `ObserverContext` might either propagate the T generic or make `AdminObserver` non-object safe.